### PR TITLE
LLM: add Ceval benchmark test.

### PR DESCRIPTION
**Pull request by lalalapotter**
## Description

initiate Ceval benchmark test, add two example evaluators: llama and qwen.

### CODE REVIEW

File: python/llm/dev/benchmark/ceval/README.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
+## C-Eval Benchmark Test
+
+C-Eval benchmark test all users to test on [C-Eval](https://cevalbenchmark.com) datasets, which is a multi-Level multi-discipline chinese evaluation suite for foundation models. It consists of 13948 multi-choice questions spanning 52 diverse disciplines and four difficulty levels. Please check [paper](https://arxiv.org/abs/2305.08322) and [github repo](https://github.com/hkust-nlp/ceval) for more information
```
Comment (hkvision) [inline python/llm/dev/benchmark/ceval/README.md:3]: test all users to test? should be allows?

Comment (hkvision) [reply python/llm/dev/benchmark/ceval/README.md:3]: Level -> level

File: python/llm/dev/benchmark/ceval/README.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+```shell
+bash run.sh
+```
++ `run.sh`
+```shell
+python eval.py \
+        --model_family qwen \
+        --model_path /mnt/disk1/models/Qwen-7B-Chat \
+        --eval_type validation \
+        --device xpu \
```
Comment (hkvision) [inline python/llm/dev/benchmark/ceval/README.md:25]: Can we support xpu:0/1?

File: python/llm/dev/benchmark/ceval/eval.py [hunk 3]
```diff
@@ -0,0 +1,308 @@
...
+#
+
+import os
+import argparse
+import pandas as pd
+import torch
+import json
+from tqdm import tqdm
+
+import intel_extension_for_pytorch as ipex
```
Comment (hkvision) [inline python/llm/dev/benchmark/ceval/eval.py:24]: why import ipex here? We should also support cpu?
Import when xpu in device?

File: python/llm/dev/benchmark/ceval/eval.py [hunk 4]
```diff
@@ -0,0 +1,308 @@
...
+            device=args.device,
+            qtype=args.qtype
+        )
+    elif args.model_family == "qwen":
+        evaluator = QwenEvaluator(
+            choices=choices,
+            model_path=args.model_path,
+            device=args.device,
+            qtype=args.qtype
+        )
```
Comment (hkvision) [inline python/llm/dev/benchmark/ceval/eval.py:306]: raise error for other families?

File: python/llm/dev/benchmark/ceval/README.md [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+        --eval_data_path data \
+        --qtype sym_int4
+```
+
+> **Note**
+>
+> `eval_type` there is two types of evaluation, first type is `validation`, which runs on validation dataset and output evaluation scores. The second type is `test`, which runs on test dataset and output `submission.json` file for submission on https://cevalbenchmark.com
```
Comment (hkvision) [inline python/llm/dev/benchmark/ceval/README.md:32]: for submission on https://cevalbenchmark.com `to get the evaluation score.`

File: python/llm/dev/benchmark/ceval/eval.py (hunk 3 above)
Comment (lalalapotter) [reply python/llm/dev/benchmark/ceval/eval.py:24]: This import can be removed, because current we will import ipex automatically.

Comment (hkvision) [reply python/llm/dev/benchmark/ceval/eval.py:24]: cool

**Review by hkvision — APPROVED** [review]

**Pull request merged**